File: widgetssdk/src/main/java/com/glia/widgets/core/fileupload/domain/SupportedFileCountCheckUseCase.java
package com.glia.widgets.core.fileupload.domain;

import com.glia.widgets.core.fileupload.FileAttachmentRepository;

public class SupportedFileCountCheckUseCase {

    private final FileAttachmentRepository repository;

    public SupportedFileCountCheckUseCase(FileAttachmentRepository repository) {
        this.repository = repository;
    }

    public boolean execute() {
        return repository.getFileAttachments().size() <= SUPPORTED_FILE_COUNT;
    }

    private final static long SUPPORTED_FILE_COUNT = 25;
}

File: widgetssdk/src/main/java/com/glia/widgets/core/fileupload/domain/SupportedFileCountCheckUseCase.kt
package com.glia.widgets.core.fileupload.domain

internal class SupportedFileCountCheckUseCase(private val fileAttachmentUseCase: ChatFileAttachmentRepositoryUseCase) {
    operator fun invoke(): Boolean {
        return fileAttachmentUseCase().getAttachedFilesCount() <= SUPPORTED_FILE_COUNT
    }

    companion object {
        const val SUPPORTED_FILE_COUNT: Long = 25
    }
}
